### PR TITLE
[install script] Adds capability to specify a minor (and patch) version

### PR DESCRIPTION
**Pull request by ian28223**
### What does this PR do?

Adds capability to specify a minor[.patch] version.

Above adds the following environment variables that can be set:

- `DD_MINOR_VERSION` - e.g. `20` or `20.0` (default: <blank>)
  - `20` will install the latest patch version e.g. `x.20.2`
  - `20.0` will install a specific patch version e.g. `x.20.0`

### Motivation

FRAGENT-963

### Additional Notes

Tested on:
- Centos 7, 8
- Ubuntu 18, 20
- Fedora 30, 33
- Amazon Linux 2
- OpenSuse 12, 15
- Debian 8, 9

### Describe how to test your changes

- invalid minor version, terminates
```
DD_AGENT_MAJOR_VERSION=7 DD_AGENT_MINOR_VERSION=99 DD_API_KEY=xxx bash -c "$(curl -H 'Cache-control: no-cache' -L https://raw.githubusercontent.com/DataDog/datadog-agent/ian.bucad/opt_min_version/cmd/agent/install_script.sh)"
DD_AGENT_MINOR_VERSION=99 DD_API_KEY=xxx bash -c "$(curl -H 'Cache-control: no-cache' -L https://raw.githubusercontent.com/DataDog/datadog-agent/ian.bucad/opt_min_version/cmd/agent/install_script.sh)"
```

- installs specified patch version, x.20.0
```
DD_AGENT_MAJOR_VERSION=7 DD_AGENT_MINOR_VERSION=20.0 DD_API_KEY=xxx bash -c "$(curl -H 'Cache-control: no-cache' -L https://raw.githubusercontent.com/DataDog/datadog-agent/ian.bucad/opt_min_version/cmd/agent/install_script.sh)"
DD_AGENT_MINOR_VERSION=20.0 DD_API_KEY=xxx bash -c "$(curl -H 'Cache-control: no-cache' -L https://raw.githubusercontent.com/DataDog/datadog-agent/ian.bucad/opt_min_version/cmd/agent/install_script.sh)"
```

- installs latest patch version e.g. x.20.2
```
DD_AGENT_MAJOR_VERSION=7 DD_AGENT_MINOR_VERSION=20 DD_API_KEY=xxx bash -c "$(curl -H 'Cache-control: no-cache' -L https://raw.githubusercontent.com/DataDog/datadog-agent/ian.bucad/opt_min_version/cmd/agent/install_script.sh)"
DD_AGENT_MINOR_VERSION=20 DD_API_KEY=xxx bash -c "$(curl -H 'Cache-control: no-cache' -L https://raw.githubusercontent.com/DataDog/datadog-agent/ian.bucad/opt_min_version/cmd/agent/install_script.sh)"
```

### CODE REVIEW

File: cmd/agent/install_script.sh [hunk 1]
```diff
@@ -6,7 +6,7 @@
 # using the package manager and Datadog repositories.
 
 set -e
-install_script_version=1.4.0
+install_script_version=1.5.0
```
Comment (mx-psi) [inline cmd/agent/install_script.sh:9]: Let's keep this as `1.4.0` and do this in a future PR when we update the changelog
```suggestion
install_script_version=1.4.0
```

File: cmd/agent/install_script.sh [hunk 2]
```diff
@@ -177,6 +197,21 @@ else
...
+if [ -n "$DD_AGENT_MINOR_VERSION" ]; then
+  # Examples:
+  #  - 20   = defaults to highest patch version x.20.2
+  #  - 20.0 = sets explicit patch version x.20.0
+  # Note: Specifying an invalid minor version will terminate the script unless DD_FALLBACK_LATEST is true
+  agent_minor_version=$DD_AGENT_MINOR_VERSION
+fi
+
+if [ -n "$DD_FALLBACK_LATEST" ]; then
+  # Used in conjuntion with DD_AGENT_MINOR_VERSION.
```
Comment (mx-psi) [inline cmd/agent/install_script.sh:209]:
```suggestion
  # Used in conjunction with DD_AGENT_MINOR_VERSION.
```

File: cmd/agent/install_script.sh [hunk 3]
```diff
@@ -310,6 +345,18 @@ if [ "$OS" = "RedHat" ]; then
       dnf_flag="--best"
     fi
 
+    if [ -n "$agent_minor_version" ]; then
+        # Example: datadog-agent-7.20.2-1
+        pkg_pattern="$agent_major_version\.${agent_minor_version%.}(\.[[:digit:]]+){0,1}(-[[:digit:]])?"
```
Comment (mx-psi) [inline cmd/agent/install_script.sh:350]: Have you tested this with RCs? I think it should work fine but we should check it

**Review by mx-psi — CHANGES_REQUESTED** [review]
Thanks for doing this! It looks good, I left some comments. You should add a release note for this describing the new feature (on the install script release notes folder)

File: cmd/agent/install_script.sh [hunk 4]
```diff
@@ -47,16 +47,21 @@ useful and we will do our very best to help you solve your problem.\n"
...
+      --data-urlencode "apikey=${apikey}" \
+      "$report_failure_url"; then
+     printf "A notification has been sent to Datadog with the contents of $logfile\n"
+    else
+      printf "Unable to send the notification (curl v7.18 or newer is required)"
+      fallback_msg
+    fi
   else
-    printf "Unable to send the notification (curl v7.18 or newer is required)"
+    printf "No email provided; skipping report.\n"
```
[the remark this answers is not in the document]
Comment (ian28223) [reply cmd/agent/install_script.sh:64]: I'll move this to another PR.

File: cmd/agent/install_script.sh [hunk 5]
```diff
@@ -177,6 +197,21 @@ else
...
+  # Note: Specifying an invalid minor version will terminate the script unless DD_FALLBACK_LATEST is true
+  agent_minor_version=$DD_AGENT_MINOR_VERSION
+fi
+
+if [ -n "$DD_FALLBACK_LATEST" ]; then
+  # Used in conjuntion with DD_AGENT_MINOR_VERSION.
+  # When an invalid value is provided for DD_AGENT_MINOR_VERSION, setting this to true will instruct the script
+  # to proceed installing the latest minor.patch version available.
+  fallback_latest=true
+fi
```
[the remark this answers is not in the document]
Comment (ian28223) [reply cmd/agent/install_script.sh:213]: I don't really have a use case for this TBH. I can take this part outm

File: cmd/agent/install_script.sh (hunk 1 above)
Comment (ian28223) [reply cmd/agent/install_script.sh:9]: I'll revert this part last after testing

File: cmd/agent/install_script.sh (hunk 3 above)
Comment (ian28223) [reply cmd/agent/install_script.sh:350]: I haven't. how do I test this with RC? are there a different apt/yum url?

Comment (mx-psi) [reply cmd/agent/install_script.sh:350]: Left you a Slack message!

File: cmd/agent/install_script.sh (hunk 5 above)
[the remark this answers is not in the document]
Comment (mx-psi) [reply cmd/agent/install_script.sh]: I think we can take it out for now, and add it in the future if the need arises

File: cmd/agent/install_script.sh (hunk 3 above)
Comment (ian28223) [reply cmd/agent/install_script.sh:336]: tested and works with RCs. Following command worked in Centos8, Fedora33, Ubuntu20, Debian9, Opensuse15

`DD_AGENT_MAJOR_VERSION=7 DD_AGENT_MINOR_VERSION=28.0~rc.7 REPO_URL=datad0g.com DD_AGENT_DIST_CHANNEL=beta DD_API_KEY=xxx bash -c "$(curl -H 'Cache-control: no-cache' -L https://raw.githubusercontent.com/DataDog/datadog-agent/ian.bucad/opt_min_version/cmd/agent/install_script.sh)"`

**Review by mx-psi — APPROVED** [review]
LGTM, thanks again! 🚀

**Pull request merged**